packages/babel-preset-expo/index.js

```diff
@@ -1,16 +1,38 @@
+const lazyImportsBlacklist = require('./lazy-imports-blacklist');
+
 module.exports = function(api, options) {
   const isWeb = api.caller(isTargetWeb);
 
   if (isWeb) {
     return getWebConfig(options.web);
   }
 
-  return getNativeConfig();
+  return getNativeConfig(options);
 };
 
-function getNativeConfig() {
+function getNativeConfig(options) {
+  // Note that if `options.lazy` is not set (i.e., `null` or `undefined`),
+  // `metro-react-native-babel-preset` will handle it.
+  const lazy_option = options && options.lazy;
   return {
-    presets: ['module:metro-react-native-babel-preset'],
+    presets: [
+      [
+        // TODO: use the actual plugin - but somehow `module:metro-react-native-babel-preset` is not up-to-date yet (need a version after this commit: https://github.com/facebook/metro/commit/23e3503dde5f914f3e642ef214f508d0a699851d)
+        require('./node_modules/metro-react-native-babel-preset'),
+        {
+          lazyImportExportTransform:
+            lazy_option === true
+              ? require_name => {
+                  // Do not lazy-initialize packages that are local imports (similar to `lazy: true` behavior)
+                  // or are in the blacklist.
+                  return !(require_name.includes('./') || lazyImportsBlacklist.has(require_name));
```

Comment:
We would like to lazily import an app's own modules (ex: App.js should lazily import `./screens/TermsOfServiceScreen`) too -- how could we enable that?

Reply:
On `@babel/plugin-transform-modules-commonjs`'s [docs](https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs#lazy) for the `lazy` option, it says
> Local paths are much more likely to have circular dependencies, which may break if loaded lazily, so they are not lazy by default, whereas dependencies between independent modules are rarely cyclical.

Also, I found that lazily importing local modules broke the Back navigation button in `apps/native-component-list/`. So I think it would be better to not lazily load local modules by default. If the developer wants to lazily import an app's own modules, they could use `(string) => boolean` to handle it themselves.

(Though if we are doing that, we should probably find a way to provide `lazyImportsBlacklist` to the developer.)

Reply:
I added a note here: https://github.com/expo/expo/pull/4685/commits/8cf5a62c416e90ff78b3bdf12e3f32da8b2b531e#diff-cc0648e61fc1cd454f9938303cc8d00d